lib/microformats2/format_parser.rb

```diff
@@ -1,60 +1,346 @@
 module Microformats2
-  class FormatParser
-    class << self
-      def parse(element, base=nil)
-        @@base = base
-        parse_node(element).flatten.compact
-      end
+  class FormatParser < ParserCore
+
+      def parse(element, base=nil, element_type=nil, fmt_classes=[], backcompat=false)
+        @base = base
+
+        @mode_backcompat = backcompat
+
+        @properties = {}
+        @children = []
+
+        @format_property_type = element_type
+        @value = nil
+
+        @mode_backcompat = backcompat
+
+        @fmt_classes =  fmt_classes
+
+        parse_node(element.children)
+
+
+        ##### Implied Properties######
+        #NOTE: much of this code may be simplified by using element.css, not sure yet, but coding to have passing tests first
+        # can optimize this later
+        unless @mode_backcompat
+          if @properties['name'].nil?
+
+            if element.name == 'img' and not element.attribute('alt').nil?
+              @properties['name'] = [element.attribute('alt').value.strip]
+            elsif element.name == 'area' and not element.attribute('alt').nil?
+              @properties['name'] = [element.attribute('alt').value.strip]
+            elsif element.name == 'abbr' and not element.attribute('title').nil?
+              @properties['name'] = [element.attribute('title').value.strip]
+
+            else
+                child_nodes = element.children.select{|n| not n.is_a?(Nokogiri::XML::Text)}
```

Comment:
Yes I had noticed this while writing it but was waiting until done to go back and fix it all.  I have vim always set to 4 spaces, but will definitely fix this